wcfsetup/install/files/lib/system/background/job/TolerantCacheRebuildBackgroundJob.class.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace wcf\system\background\job;
+
+use wcf\system\cache\CacheHandler;
+use wcf\system\cache\tolerant\AbstractTolerantCache;
+
+/**
+ * @author Olaf Braun
+ * @copyright 2001-2025 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.2
+ */
+final class TolerantCacheRebuildBackgroundJob extends AbstractUniqueBackgroundJob
+{
+    public function __construct(
+        /** @var class-string<AbstractTolerantCache<array|object> */
+        public readonly string $cacheClass,
+        /** @var array<string, mixed> */
+        public readonly array $parameters = []
+    ) {
+    }
+
+    public function identifier(): string
+    {
+        $identifier = $this->cacheClass;
+        if (!empty($this->parameters)) {
+            $identifier .= '-' . CacheHandler::getInstance()->getCacheIndex($this->parameters);
+        }
+
+        return $identifier;
+    }
+
+    #[\Override]
+    public function newInstance(): static
+    {
+        return new TolerantCacheRebuildBackgroundJob($this->cacheClass, $this->parameters);
+    }
+
+    #[\Override]
+    public function queueAgain(): bool
+    {
+        return \class_exists($this->cacheClass);
```

Comment:
This kind of defeats the purpose of the tolerant cache principle because the `BackgroundQueueHandler` would queue it for execution in `retryAfter()` seconds. I don’t think it should ever re-queue itself and instead rely on being asked to refresh itself when necessary.

The whole idea is that this cache could go stale but still being served while it would be put at the end of the queue to be refreshed (unless it already exists of course).

The difference is that your current approach actively queues this cache for a rebuild, while always return `false` makes this an opt-in approach instead. This reduces the amount of times the cache is being rebuilt (allowing it to go stale more often) and reduces the load on the background queue.